### PR TITLE
ReST file lint

### PR DESCRIPTION
**Pull request by leogermond**
Using my own linter I was able to fix the following issues:

+ trailing whitespace
+ trailing list-element whitespace
+ text outside slide
+ blank line
+ blank line in a list
+ malformed code block
+ missing line-jump between lists

### CODE REVIEW

**Comment by leogermond** [comment]
> I'm assuming the PDF looks the same :)
> (You never know with Pandoc!)

Yes I used the linted version for presentation, it looked exactly the same, the linter is quite conservative and it should mostly be helpful for eg merge when indentation has been fixed somewhere: we can fix it in both branches and the fake conflict disappears

File: courses/fundamentals_of_ada/070_subprograms.rst [hunk 1]
```diff
@@ -1034,13 +1031,13 @@ Return Statements In Functions
...
         end designator;
- 
+
 ---------------------------------------
 No Path Analysis Required By Compiler
 ---------------------------------------
 
-* Running to the end of a function without hitting a `return` statement raises `Program_Error`
-* Compilers can issue warning if they suspect that a `return` statement will not be hit
+* Running to the end of a function without hitting a :ada:`return` statement raises `Program_Error`
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/070_subprograms.rst:1039]: Program_Error should be using :ada:

File: courses/fundamentals_of_ada/070_subprograms.rst [hunk 2]
```diff
@@ -1382,7 +1379,7 @@ Order-Dependent Code And Side Effects
    end Gear_Down;
    ...
    Gear_Down (Global, F);
-     
+
 * Order of evaluation of parameters in subprogram call is not specified in language
 * `Gear_Down` could get called with
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/070_subprograms.rst:1384]: Gear_Down should be using :ada:

File: courses/fundamentals_of_ada/030_basic_types.rst [hunk 3]
```diff
@@ -327,14 +326,14 @@ Integer Overflows
    K : Short_Integer := Short_Integer'Last;
    ...
    K := K + 1;
-      
+
     2#0111_1111_1111_1111#  = (2**16)-1
 
-   +                    1
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/030_basic_types.rst:333]: Spacing is important - convert to a literal block?

**Review by frank-at-adacore — CHANGES_REQUESTED** [review]
Bunch of spaces where we still don't use :ada: - you can fix or leave them for next time
2 cases where we were using spaces as an indentation - they should be fixed

File: courses/fundamentals_of_ada/050_array_types.rst [hunk 4]
```diff
@@ -942,9 +939,9 @@ For-Loops with Multidimensional Arrays
     * In low-level format looks like
 
        - for each row loop
-       -   for each column loop
-       -     print Identity (row, column)
-       -   end loop
+       - for each column loop
```
[the remark this answers is not in the document]
Comment (leogermond) [reply courses/fundamentals_of_ada/050_array_types.rst:942]: Yes let's do this

File: courses/fundamentals_of_ada/430_ada_text_io.rst [hunk 5]
```diff
@@ -211,9 +209,8 @@ Ada.Text_IO.Float_IO
 
    - **12** will set F to 12.0
    - **23.45.67** will set F to 23.45
-      
-* `IO` has global objects `Default_Fore`, `Default_Aft` and `Default_Exp` which can be modified to set default values for like-named parameters
 
+* `IO` has global objects `Default_Fore`, `Default_Aft` and `Default_Exp` which can be modified to set default values for like-named parameters
```
[the remark this answers is not in the document]
Comment (leogermond) [reply courses/fundamentals_of_ada/430_ada_text_io.rst:213]: Ok ! I did let the tool make the choice, and for ambiguous cases like this one where there is probably no syntax coloring happening it doesnt change the role

File: courses/fundamentals_of_ada/030_basic_types.rst (hunk 3 above)
Comment (leogermond) [reply courses/fundamentals_of_ada/030_basic_types.rst:333]: Indeed

Comment (leogermond) [reply courses/fundamentals_of_ada/030_basic_types.rst:333]: Fixed the linter ! :)

**Comment by leogermond** [comment]
> Bunch of spaces where we still don't use :ada: - you can fix or leave them for next time
> 2 cases where we were using spaces as an indentation - they should be fixed

I added a condition on the linter, as for adding those manually that could be the next step, but it would require to eg have something like a defaut `:pre:` role to any "blank" code block so that the linter doesnt complain

**Review by frank-at-adacore — APPROVED** [review]

**Pull request merged**